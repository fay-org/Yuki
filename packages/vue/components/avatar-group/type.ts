import type { NormalSizes } from '../utils'

export const READONLY_AVATAR_KEY = Symbol('avatarKey')

export interface AvatarGroupProvide {
  props: {
    stacked: boolean
    isSquare: boolean
    size: NormalSizes
  }
}
