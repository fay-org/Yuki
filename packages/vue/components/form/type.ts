import { ComputedRef } from 'vue'
import { tuple, NormalSizes } from '../utils'
import type { FormProps } from './props'

export const labelPosition = tuple('left', 'right', 'top')

export type LabelPosition = typeof labelPosition[number]

export const READONLY_FORM_KEY = Symbol('formKey')

export type FormProvide = {
  formProps: FormProps
  getLabelPostion: (postion: LabelPosition | '') => Exclude<LabelPosition, 'top'> | null
}
