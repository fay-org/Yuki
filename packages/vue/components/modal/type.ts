import { Ref, ExtractPropTypes } from 'vue'
import { props, staticModalProps } from './props'

export type ModalProvide = {
  props: ExtractPropTypes<typeof props>
  selfVisible: Ref<boolean>
  setSelfVisible: (val: boolean) => void
}

export const READONLY_MODAL_KEY = 'modalKey'

export type StaticModalOptions = ExtractPropTypes<typeof staticModalProps> & {
  confirm: () => void
  close: () => void
}
