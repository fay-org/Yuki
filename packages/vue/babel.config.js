module.exports = {
  presets: ['@fect-ui/cli/preset'],
}
