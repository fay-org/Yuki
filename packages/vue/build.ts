import { BuildTaskConfig, TASK_NAME, runTask, commonOutput } from 'internal'
import { createBundle } from 'no-bump'

const { build } = createBundle({})

const commonInput = 'src/index.ts'

/**
 * I decide split the build task into four.
 * First is only bundle script module.
 * Second part only parser style moudle.
 * Thrid part will generator all analyzed and generator style deps graph.
 * Fourth part will generator the entry file as umd.
 *
 * The first task will have two sub task. with esm task coomonjs task
 */
