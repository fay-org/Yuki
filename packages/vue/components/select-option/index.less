.fect-option {
  display: flex;
  max-width: 100%;
  justify-content: flex-start;
  align-items: center;
  font-size: var(--select-font-size);
  height: calc(1.688 * var(--fect-gap));
  padding: 0 var(--fect-gap-half);
  background-color: var(--primary-background);
  color: var(--accents-5);
  user-select: none;
  border: 0;
  cursor: pointer;
  &:not(&--disabled):hover {
    background-color: var(--accents-1);
    color: var(--primary-foreground);
  }
  &--disabled {
    cursor: not-allowed;
    color: var(--accents-3);
    background-color: var(--accents-1);
  }
}
