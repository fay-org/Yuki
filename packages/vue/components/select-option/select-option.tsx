import { defineComponent } from 'vue'
import { useSelectContext } from '../select/select-context'
import { selectOptionProps } from '../select/props'
import { createName, createBem } from '../utils'
import './index.less'

const name = createName('Option')
const bem = createBem('fect-option')

export default defineComponent({
  name,
  props: selectOptionProps,
  setup(props) {
    const { context } = useSelectContext()
    const { setVisible, updateSelectValue } = context!

    const handleClick = (e: Event) => {
      e.stopPropagation()
      e.preventDefault()
      if (props.disabled) return
      setVisible(false)
      updateSelectValue(props.value)
    }
    return () => (
      <div class={bem(null, { disabled: props.disabled })} onClick={handleClick}>
        {props.label}
      </div>
    )
  }
})
