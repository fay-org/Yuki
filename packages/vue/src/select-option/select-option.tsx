import { defineComponent, computed } from 'vue'
import { useSelectContext } from '../select/select-context'
import { selectOptionProps } from '../select/props'
import { createName, createBem } from '../utils'
import './index.less'

const name = createName('Option')
const bem = createBem('fect-option')

export default defineComponent({
  name,
  props: selectOptionProps,
  setup(props) {
    const { context } = useSelectContext()
    const { updateSelectVisible, updateSelectValue, updateDropDown, parentValue, selectState } = context!

    const handleClick = (e: Event) => {
      e.stopPropagation()
      e.preventDefault()
      if (props.disabled) return
      updateSelectVisible()
      updateSelectValue(props.value)
      updateDropDown()
    }

    const setOptionClass = computed(() => {
      const parentVal = parentValue.value
      const checked = Array.isArray(parentVal) ? parentVal.includes(props.value as string) : parentVal === props.value
      return bem(null, { disabled: props.disabled, size: selectState.value.size, checked })
    })

    return () => (
      <div class={setOptionClass.value} onClick={handleClick}>
        {props.label}
      </div>
    )
  }
})
