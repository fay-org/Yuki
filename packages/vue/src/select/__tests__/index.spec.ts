import { mount } from '@vue/test-utils'
import { Select } from '..'
import { nextTick } from 'vue'
import type { SelectProps } from '../interface'
import { SelectOption } from '../../select-option'
import { later, trigger } from '../../../tests'

const _wrapper = (prop?: SelectProps) => {
  return mount({
    components: {
      [SelectOption.name]: SelectOption,
      [Select.name]: Select
    },
    data() {
      return {
        val: '' || prop?.modelValue,
        placeholder: 'mock placeholder test',
        multiple: prop?.multiple || false,
        clearable: true,
        disabled: prop?.disabled || false,
        size: prop?.size || 'medium',
        childrenDisabled: false
      }
    },
    template: `
        <div class="container">
         <fe-select
         ref="selectRef"
         v-model="val"
         :multiple="multiple"
         :placeholder="placeholder"
         :disabled="disabled"
         :clearable="clearable"
         :size="size"
         :data-size="size"
         >
           <fe-option label="吃饭" value="1"></fe-option>
           <fe-option label="摸鱼" value="2" :disabled="childrenDisabled"></fe-option>
         </fe-select>
        </div>
        `
  })
}

describe('Select', () => {
  it('should wrork correctly', async () => {
    const wrapper = _wrapper({ size: 'mini' })
    expect(wrapper.find('.fect-select--mini').exists()).toBe(true)
    await wrapper.trigger('click')
    const option = document.querySelector('.fect-option')!
    await trigger('click', option)
    expect(wrapper.vm.val).toBe('1')
    const contentEl = wrapper.find('.fect-select__content')
    await contentEl.trigger('mouseenter')
    expect(wrapper.find('.fect-select__arrow--clear').exists()).toBe(true)
    await contentEl.trigger('mouseleave')
    expect(wrapper.find('.fect-select__arrow--clear').exists()).toBe(false)
    await contentEl.trigger('mouseenter')
    await wrapper.find('.fect-input__icon--suffix').trigger('click')
    expect(wrapper.vm.val).toBe('')
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should support render with multiple mode', async () => {
    const wrapper = await _wrapper({ multiple: true, modelValue: '2' })
    const itemIconEl = wrapper.find('.fect-select__item svg')
    await itemIconEl.trigger('click')
    expect(wrapper.vm.val?.length).toBe(0)
    await wrapper.setData({ childrenDisabled: true })
    await wrapper.trigger('click')
    const option = document.querySelector('.fect-option')!
    await trigger('click', option)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
