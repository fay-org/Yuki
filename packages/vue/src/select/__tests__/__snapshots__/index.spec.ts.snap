// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should support render with multiple mode 1`] = `
<div class="container">
  <div class="fect-select fect-select--medium" style="--select-content-height: 0px;" data-size="medium">
    <div class="fect-tooltip">
      <div class="fect-select__content">
        <div class="fect-grid__group fect-grid__group--wrap fect-select__multiple" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start; --fect-grid-gap: calc(0.5 * 16px * 1/3);"></div>
        <div class="fect-input fect-input--medium">
          <!---->
          <div class="fect-input__container">
            <!---->
            <div class="fect-input__wrapper">
              <!----><input type="text" class="fect-select__input" readonly="" placeholder="mock placeholder test" role="combobox" aria-haspopup="listbox" aria-expanded="false">
              <!---->
              <div class="fect-input__icon fect-input__icon--suffix" style="cursor: pointer; pointer-events: auto;">
                <div class="fect-select__arrow"><svg viewBox="0 0 24 24" height="20" width="20" stroke-width="1" stroke-linecap="round" color="currentColor" stroke="currentColor" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
                    <path d="M6 9l6 6 6-6"></path>
                  </svg></div>
              </div>
            </div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
</div>
`;

exports[`Select should wrork correctly 1`] = `
<div class="container">
  <div class="fect-select fect-select--mini" style="--select-content-height: calc(var(--select-ratio) * var(--fect-gap));" data-size="mini">
    <div class="fect-tooltip">
      <div class="fect-select__content">
        <!---->
        <div class="fect-input fect-input--mini">
          <!---->
          <div class="fect-input__container">
            <!---->
            <div class="fect-input__wrapper">
              <!----><input type="text" class="fect-select__input" readonly="" placeholder="mock placeholder test" role="combobox" aria-haspopup="listbox" aria-expanded="false">
              <!---->
              <div class="fect-input__icon fect-input__icon--suffix" style="cursor: pointer; pointer-events: auto;">
                <div class="fect-select__arrow"><svg viewBox="0 0 24 24" height="20" width="20" stroke-width="1" stroke-linecap="round" color="currentColor" stroke="currentColor" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
                    <path d="M6 9l6 6 6-6"></path>
                  </svg></div>
              </div>
            </div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
</div>
`;
